python_modules/dagster/dagster/_core/definitions/load_assets_from_modules.py

```diff
@@ -402,10 +407,11 @@ def replace_keys_in_asset(
     asset: Union[AssetsDefinition, SourceAsset],
     key_replacements: Mapping[AssetKey, AssetKey],
 ) -> Union[AssetsDefinition, SourceAsset]:
-    return (
+    updated_object = (
         asset.with_attributes(
             output_asset_key_replacements={
-                key: key_replacements.get(key, key) for key in asset.keys
+                key: key_replacements.get(key, key)
+                for key in key_iterator(asset, included_targeted_keys=True)
```

Comment:
unclear to me why we would be including targeted keys in this iteration -- all of the node's output asset keys should already be captured by `asset.keys`, right?

Reply:
looking at the original code, should it be `input_asset_key_replacements` that gets this treatment?

Reply:
see original code: https://github.com/dagster-io/dagster/blob/279d44aa2f325f2d2a883807da6f29bf3a2b10e1/python_modules/dagster/dagster/_core/definitions/load_assets_from_modules.py#L422

output asset keys does _not_ output the keys targeted by checks. We need to explicitly use the set operator to add them to the list of output keys we process.

It's actually `input_asset_key_replacements` which automatically handles checks; which is why we don't have to add any additional handling here.

A good followup would be to move these methods either into the AssetsDefinition itself or at least have a kind of single place for this complexity on AssetsDefinition I think.